bluetoe/link_layer/include/bluetoe/peripheral_latency.hpp

```diff
@@ -0,0 +1,236 @@
+#ifndef BLUETOE_LINK_LAYER_PERIPHERAL_LATENCY_HPP
+#define BLUETOE_LINK_LAYER_PERIPHERAL_LATENCY_HPP
+
+#include <bluetoe/ll_meta_types.hpp>
+
+/**
+ * @file bluetoe/peripheral_latency.hpp
+ *
+ * This file provides several options, how Bluetoe will implement peripheral latency. Peripheral latency
+ * allows a link layer peripheral to miss some connection events (do not respond to the central) to
+ * conserve power.
+ *
+ * Each option is a compromise between latency and power consumption.
+ *
+ * @sa bluetoe::link_layer::peripheral_latency_ignored
+ * @sa ...
+ */
+namespace bluetoe {
+namespace link_layer {
+
+    namespace details {
+        struct peripheral_latency_meta_type {};
+    }
+
+    /**
+     * @brief detailed options for the peripheral latency behavior
+     *
+     * Every option defines a set of events / circumstances under which
+     * the link layer should listen to the very next connection event.
+     * (beside the miniumum events, the link layer has to listen to due
+     * to the periperhal latency value of the connection).
+     *
+     * @note in combination, there are some options that make no sense.
+     *       For example, using listen_always with some of the other
+     *       listen_* options is pointless. Also combining send_always
+     *       with send_pending_data doesn't make sense.
+     */
+    enum class peripheral_latency
+    {
+        /**
+         * @brief listen at all connection events, despite the current
+         *        peripheral latency value of the current connection.
+         *
+         * This does not imply, that the link layer will also respond at every
+         * connection event. Just listening, without responding will
+         * synchronize the link layer with the central and thus narrow the
+         * receive window size at the next connection interval.
+         */
+        listen_always,
+
+        /**
+         * @brief
+         */
+        listen_if_pending_data,
+
+        /**
+         * @brief listen, if the link layer contains unacknowledged data.
+         *
+         * If a PDU was send out, but no acknowledgement was jet received,
+         * the link layer will listen at the next connection event, if
+         * this options is set.
+         *
+         * This might be interesting, if the sending queue is very small because
+         * Bluetoe has to keep a copy of a transmitted PDU, as long as that PDU
+         * is not acknowledged.
```

Comment:
As soon, as at N+X, the central acknowledgement was received, the slave would not reply with [S Same Data PDU], but with the next PDU. The master will repeat its Ack at N+1, N+2 and N+X by still sending the expected `next sequence number` NESN. 

"When resending a Data Physical Channel PDU, the LLID, SN, and CP fields, the CTEInfo field (if present), and the payload of the sent Data Physical Channel PDU shall be equal to those of the last Data Physical Channel PDU sent by the Link Layer."

This doesn't include the NESN, so the central can acknowledge receiving data even when it's own, send out data, will not be acknowledge by the peripheral. This happens for example, when the peripheral correctly received the centrals PDU, but was not able to store it at the moment.

Reply:
Oups sorry yes you are right. I was out of my mind; I reckon this is a side effect we had in BR/EDR but that was addressed in LE precisely by using SN/NESN instead of ACK/NAK.